### ISSUE TITLE
Fix CreateScreens.t in rel-10_1, and in rel-11_0 with S3 activated

### ISSUE THREAD

**Issue by bschmalhofer** [RotherOSS/otobo#1067]
Looks better now, down to a single failure. Looks like some kind of layout issue. Consistenly, the same failure shows in #1032.
```

# Failed test 'RunTest: no exception'
# at /opt/otobo/Kernel/System/UnitTest/Selenium.pm line 306.
# Exception: Error while executing command: element click intercepted: element click intercepted: Element <a href="#" class="AttachmentDelete" tabindex="0" aria-label="Click to delete this attachment.">...</a> is not clickable at point (948, 1066). Other element would receive the click: <div id="Overlay" tabindex="-1"></div>
#   (Session info: chrome=90.0.4430.85) at /opt/otobo_install/local/lib/perl5/Selenium/Remote/Driver.pm line 411.
#  at /opt/otobo_install/local/lib/perl5/Selenium/Remote/Driver.pm line 361.
# Looks like you failed 1 test of 134.

/opt/otobo/scripts/test/Selenium/Agent/MultiAttachmentUpload/CreateScreens.t                              (Wstat: 256 Tests: 134 Failed: 1)
  Failed test:  134
  Non-zero exit status: 1

```

_Originally posted by @bschmalhofer in https://github.com/RotherOSS/otobo/issues/1011#issuecomment-857572121_

**Comment by bschmalhofer** [comment]
Looks like this error either has gone away or is sporadic. Closing this issue.

**Comment by bschmalhofer** [comment]
Looks like this error cropped up again in rel-10_1: 
![1604007966100000_1](https://user-images.githubusercontent.com/46019/142039037-0a943919-ea16-41f9-ae94-4e751cfcf9e9.png)

**Comment by bschmalhofer** [comment]
Reliably seen with rel-10_1 with S3 backend active. Reopening.

**Comment by bschmalhofer** [comment]
Still seeing this failure occasionally. Let's keep this issue open.

**Comment by bschmalhofer** [comment]
Tried this again with rel-11_0 in a Docker container and activated S3 support. I ran 
`prove -v -I . -I Kernel/cpan-lib/ scripts/test/Selenium/Agent/MultiAttachmentUpload/CreateScreens.t` several times and got at least once an error. Looks like a timing issue.

Let's keep this issue still open.

**Comment by bschmalhofer** [comment]
Added another wait step for fixing a timing issue. But that does not address the "uploaded again" issue. It looks like the list of uploaded attachments, as seen by JavaScript, still contains attachments that already have been deleted. Hence there is another overlay which prohibits that the waste basket can be clicked.

**Comment by bschmalhofer** [comment]
Occurs also in rel-11_0 without S3:

```
# Failed test 'RunTest: no exception'
# at /opt/otobo/Kernel/System/UnitTest/Selenium.pm line 263.
# Exception: Error while executing command: element click intercepted: element click intercepted: Element <a href="#" class="AttachmentDelete" tabindex="0" aria-label="Click to delete this attachment.">...</a> is not clickable at point (948, 1066). Other element would receive the click: <div id="Overlay" tabindex="-1"></div>
#   (Session info: chrome=90.0.4430.85) at /opt/otobo_install/local/lib/perl5/Selenium/Remote/Driver.pm line 411.
#  at /opt/otobo_install/local/lib/perl5/Selenium/Remote/Driver.pm line 361.
# Looks like you failed 1 test of 136.
[16:28:29] /opt/otobo/scripts/test/Selenium/Agent/MultiAttachmentUpload/CreateScreens.t ................................ 

```

**Comment by bschmalhofer** [comment]
I tried to understand why there are sporadic failures by running the test script and watching the remote controlled browser with Remmina. However, I could never discern which the trash bin was not clickable. Now, let's go with a trial and error approach, and switch the XPath to a CSS selector. Let's see whether the error occurs again.

**Comment by bschmalhofer** [comment]
Got the same error again when running with rel-11_0 and S3 activated. Let's try with readding a 'sleep 1'.

**Comment by bschmalhofer** [comment]
I investigated where the Overlay that prevents clicking on the trashbin comes from. The Selenium error message `<div id="Overlay" tabindex="-1">` indicates the funktion `ShowDialog()`in _var/httpd/htdocs/js/Core.UI.Dialog.js_ . There the Overlay is added for modal dialogs. This finding can be proven by adding a dummy attribute in _Core.UI.Dialog.js_. This dummy attribute then shows up in the error message.

This suggests that something fishy is going on when closing modal dialogs. Either `CloseDialog()`is never called, or it does not remove the Overlay. Commenting out
https://github.com/RotherOSS/otobo/blob/84db3fdb73277ddc8b139839cc77eba03ca4a0f6/var/httpd/htdocs/js/Core.UI.Dialog.js#L716 reliably produces the error.

After this issue is resolved, the various "fixes" in _CreateScreens.t_" can be removed again.

**Comment by stefanhaerter** [comment]
I managed to examine the following:

- The error messages, overlays etc. are working as expected
- The error occurs reliably after a container restart (`docker restart otobo_web_1`)
- The error message in the screenshot in https://github.com/RotherOSS/otobo/issues/1067#issuecomment-970519082 is correct, because when the error occurs, Text1.pdf has already been uploaded successfully and the next event tries to upload Text1.doc and Text1.pdf together, which produces the error

**Comment by stefanhaerter** [comment]
Should be fixed with #2292

**Comment by bschmalhofer** [comment]
The latest runs of the test suite showed no errors. Looks like the error in _CreateScreens.t_ is gone. Closing this issue.
